### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vova-Stoliar**
[DEMO LINK](https://Vova-Stoliar.github.io/react_list-of-goods/)

### CODE REVIEW

**Review by set1one — APPROVED** [review]
Great!

File: src/App.tsx [hunk 1]
```diff
@@ -1,5 +1,8 @@
+/* eslint-disable no-console */
```
Comment (danheim) [inline src/App.tsx:1]: You are not allowed to do this

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +17,167 @@ const goodsFromServer: string[] = [
...
+      reset,
+      reverseGoods,
+      maxWordLength,
+    } = this.state;
+
+    let filterGoods = changedGoods;
+
+    if (+maxWordLength >= 0) {
+      filterGoods = changedGoods.filter(good => good.length <= +maxWordLength);
+    }
```
Comment (danheim) [inline src/App.tsx:122]: avoid filtering directly in render

File: src/button/button.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-console */
```
Comment (danheim) [inline src/button/button.tsx:1]: again

File: src/select/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+export const Select: React.FC<Props> = React.memo((props) => {
+  const { wordsLength } = props;
+
+  return (
+    <>
+      <option value="Word length" disabled>
+        Word length
+      </option>
+      {wordsLength.map(wordLength => {
+        return (
```
Comment (danheim) [inline src/select/Select.tsx:16]: avoid return

**Review by danheim — CHANGES_REQUESTED** [review]
good start

**Review by danheim — APPROVED** [review]

**Pull request closed without merging**